app/models/publication.rb

```diff
@@ -335,11 +310,7 @@ def update_any_new_contribution_info_in_pub_hash_to_db
         hash_for_update[:cap_profile_id] = author.cap_profile_id if author.cap_profile_id.present?
         contrib = contributions.where(author_id: author.id).first_or_initialize
         contrib.assign_attributes(hash_for_update)
-        if contrib.persisted?
-          contrib.save
-        else
-          contributions << contrib unless contributions.include? contrib
-        end
+        contrib.save if contrib.changed?
       end
```

Comment:
Some of this logic avoids duplicating contributions in the pub_hash, a past bug that was not easy to find and fix.  It's difficult to see how this change impacts that behavior, it's important that any prior bug fixes are not broken by the change.

Reply:
Right.  I think I preserved everything we want out of this.  Obviously, by definition, if the object isn't changed then there is nothing (new or edited) to save.

Reply:
The bit that has always bugged me is the
```
contributions << contrib unless contributions.include? contrib
```
IIRC, I added the `unless contributions.include? contrib` to fix a dup bug ages ago when I first started work on sul_pub.

Reply:
I get that.  Since that time we started using `contributions.where(author_id: author.id).first_or_initialize`, so there are now two possibilities:
- If we have a match, it would already be in the association, so we are updating instead of inserting
- If we built a new contrib, it is *also* already in the association.  It can't be a dup because it **must** have a different author_id.

I think the workaround was there because duplication happened when new objects (i.e. `contrib.persisted? == false`) were manually being added to the association.  Now we avoid manually inserting and ActiveRecord handles it.  If you can imagine a test case for `update_any_new_contribution_info_in_pub_hash_to_db` that would trigger duplication, I'd be all for it though!